submissions/DmitryHniezdilov/task-js-dom/js/main.js

```diff
@@ -0,0 +1,111 @@
+"use strict";
+
+import { dataList, defaultContentItem } from "./data.js";
+
+const elemLogo = document.querySelector(".js-elem-logo");
+const elemMenu = document.querySelector(".js-elem-menu");
+const elemContent = document.querySelector(".js-elem-content");
+
+const createMenuMarkup = (data) => {
+  const menuListItems = data.map((item) => {
+    return `
+                <li class="menu__item" >
+                    <button class="menu__btn" type="button" data-item-id="${item["id"]}">${item["Title"]}</button>
+                </li>
+            `;
+  });
+
+  return `
+            <ul class="menu__list">
+                ${menuListItems}
+            </ul>
```

Comment:
In this line, you put an array of strings inside DOM element, [read this instruction](https://kottans.org/documentation/docs/doc/code-review#dom-api) on how to improve this code

Reply:
Really,  I didn't understand why adding elements to the DOM from a loop was a bad practice until I checked the markup:
https://i.imgur.com/Cmmu8M9.png.
In this case I changed to appendChild method and in createСontentMarkup func I change detailsList loop to createDetailsList  func for return string.

Reply:
All those commas in markup because you pass an array of strings in DOM element)
Operations with DOM are very expensive and therefore should be minified.

Reply:
THX for the detailed description of the problem.